src/System.Reflection.Metadata/tests/TestUtilities/LoaderUtilities.cs

```diff
@@ -33,7 +34,18 @@ public static void LoadPEAndValidate(byte[] peImage, Action<PEReader> validator,
                 }
             }
 
-            File.Delete(tempFile);
+            for (int i = 0; i < 10; i++)
```

Comment:
We have a TempFile class in Common that we use elsewhere in tests:
https://github.com/dotnet/corefx/blob/d0dc5fc099946adc1035b34a8b1f6042eddb0c75/src/Common/tests/System/IO/TempFile.cs#L11
We could use it here, too.  It doesn't retry deletes, but we could change it to do so and then just use that here (or just use it here and accept that now and again a temp file will be left behind, assuming we can't fix the underlying issue).